### PR TITLE
jnp.argmin/max: correctly handle out argument

### LINKED ISSUE #14904: `argmax` and `argmin` doesn't raise when the argument `out` is used.
### Description

https://github.com/google/jax/blob/3fbe4e8230c60e40a1c54732d879438b9d39bd16/jax/_src/numpy/lax_numpy.py#L3401-L3410

`out` is never passed to `_argmax` where the error handling is actually implemented. I guess wrapped functions like `_argmax` shouldn't have default arguments.

Also, `keepdims` should be key word only.

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #14904

### CODE REVIEW

**Review by hawkinsp — APPROVED** [review]

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -3399,14 +3399,15 @@ def argwhere(a, *, size=None, fill_value=None):
 
 
 @util._wraps(np.argmax, skip_params=['out'])
-def argmax(a, axis: Optional[int] = None, out=None, keepdims=None):
-  return _argmax(a, None if axis is None else operator.index(axis), keepdims=bool(keepdims))
-
-@partial(jit, static_argnames=('axis', 'keepdims'), inline=True)
-def _argmax(a, axis: Optional[int] = None, out=None, keepdims=False):
+def argmax(a: ArrayLike, axis: Optional[int] = None, out=None, keepdims=None) -> Array:
```
Comment (soraros) [inline jax/_src/numpy/lax_numpy.py:3402]: `keepdimes` should have been keyword only.

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -3399,14 +3399,15 @@ def argwhere(a, *, size=None, fill_value=None):
...
   util._check_arraylike("argmax", a)
   if out is not None:
     raise NotImplementedError("The 'out' argument to jnp.argmax is not supported.")
+  return _argmax(asarray(a), None if axis is None else operator.index(axis),
+                 keepdims=bool(keepdims))
+
+@partial(jit, static_argnames=('axis', 'keepdims'), inline=True)
+def _argmax(a: Array, axis: Optional[int] = None, keepdims: bool = False) -> Array:
```
Comment (soraros) [inline jax/_src/numpy/lax_numpy.py:3410]: Maybe we could drop the default arguments from the inner functions, WDYT?

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:3410]: Better would be to drop the inner functions entirely... they're mostly left-over from a time when the functions returned by `jit` didn't support keyword arguments.

Comment (soraros) [reply jax/_src/numpy/lax_numpy.py:3410]: They do have a benefit, for `partial` doesn't work well with static type-checkers.

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:3410]: Yeah, I'm of two minds on that. It's nice to write code that's compatible with static type checkers, but if we're forced to write code in a confusing/inefficient way to appease them, I'd lean towards not jumping through those hoops.

Comment (jakevdp) [reply jax/_src/numpy/lax_numpy.py:3410]: We may end up just making a `pyi` file for `jax.numpy` due to issues like this.

Comment (soraros) [reply jax/_src/numpy/lax_numpy.py:3410]: > We may end up just making a pyi file for jax.numpy due to issues like this.

That's actually great news. And all the overloads will have a place to go.

**Pull request merged**